--- mozilla/mail/app/Makefile.in.orig	2009-07-06 14:29:36.000000000 +0200
+++ mozilla/mail/app/Makefile.in	2009-07-06 14:28:42.000000000 +0200
@@ -135,16 +135,17 @@ endif
 
 LIBS += \
 	$(STATIC_COMPONENTS_LINKER_PATH) \
 	$(MOZ_JS_LIBS) \
 	$(EXTRA_DSO_LIBS) \
 	$(APP_XPCOM_LIBS) \
 	$(NSPR_LIBS) \
 	$(TK_LIBS) \
+	$(MOZ_HUNSPELL_LIBS) \
 	$(NULL)
 
 # Add explicit X11 dependency when building against X11 toolkits
 ifneq (,$(filter gtk gtk2 xlib,$(MOZ_WIDGET_TOOLKIT)))
 LIBS += $(XLDFLAGS) $(XLIBS)
 endif
 
 ifdef MOZ_JPROF
