From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Tue, 5 Aug 2014 20:47:04 +0200
Subject: FTBFS hurd: adding GNU/Hurd to the list of OS systems

---
 mozilla/build/gyp.mozbuild | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mozilla/build/gyp.mozbuild b/mozilla/build/gyp.mozbuild
index 2ae5ac0..a2c4cf7 100644
--- a/mozilla/build/gyp.mozbuild
+++ b/mozilla/build/gyp.mozbuild
@@ -103,6 +103,7 @@ flavors = {
     'FreeBSD': 'freebsd',
     'NetBSD': 'netbsd',
     'OpenBSD': 'openbsd',
+    'GNU': 'gnu',
 }
 gyp_vars['OS'] = flavors.get(os)
 
