From: Mike Hommey <glandium@debian.org>
Date: Sat, 8 Dec 2007 19:24:40 +0100
Subject: Set javascript.options.showInConsole

---
 mozilla/modules/libpref/init/all.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mozilla/modules/libpref/init/all.js b/mozilla/modules/libpref/init/all.js
index 7078c66..07bb947 100644
--- a/mozilla/modules/libpref/init/all.js
+++ b/mozilla/modules/libpref/init/all.js
@@ -1153,6 +1153,7 @@ pref("javascript.options.ion.offthread_compilation", true);
 // memory, but makes things like Function.prototype.toSource()
 // fail.
 pref("javascript.options.discardSystemSource", false);
+pref("javascript.options.showInConsole",    true);
 // This preference limits the memory usage of javascript.
 // If you want to change these values for your device,
 // please find Bug 417052 comment 17 and Bug 456721
