From: Christoph Goehre <chris@sigxcpu.org>
Date: Fri, 17 Jan 2014 20:18:32 -0500
Subject: use system jquery/jquery-ui

Closes: #735234
---
 mail/jquery/jar.mn | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mail/jquery/jar.mn b/mail/jquery/jar.mn
index adc2811..d4e40a5 100644
--- a/mail/jquery/jar.mn
+++ b/mail/jquery/jar.mn
@@ -3,5 +3,5 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 messenger.jar:
-    content/messenger/jquery.js                       (jquery-1.6.4.min.js)
-    content/messenger/jquery-ui.js                    (jquery-ui-1.8.2.min.js)
+    content/messenger/jquery.js                       (jquery.min.js)
+    content/messenger/jquery-ui.js                    (jquery-ui.min.js)
