From: Mike Hommey <mh@glandium.org>
Date: Sun, 18 Mar 2012 09:37:00 +0100
Subject: Include cstdlib in gfx/angle/src/compiler/Types.h for abort()

---
 mozilla/gfx/angle/src/compiler/Types.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/mozilla/gfx/angle/src/compiler/Types.h b/mozilla/gfx/angle/src/compiler/Types.h
index 00af98b..d3152cd 100644
--- a/mozilla/gfx/angle/src/compiler/Types.h
+++ b/mozilla/gfx/angle/src/compiler/Types.h
@@ -9,6 +9,7 @@
 
 #include "common/angleutils.h"
 
+#include <cstdlib>
 #include "compiler/BaseTypes.h"
 #include "compiler/Common.h"
 #include "compiler/compiler_debug.h"
