From: Christoph Goehre <chris@sigxcpu.org>
Date: Mon, 16 Sep 2013 20:40:57 +0200
Subject: Load-dependent-libraries-with-their-real-path-to-avo

---
 mozilla/xpcom/glue/standalone/nsXPCOMGlue.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mozilla/xpcom/glue/standalone/nsXPCOMGlue.cpp b/mozilla/xpcom/glue/standalone/nsXPCOMGlue.cpp
index 578c706..f4f1568 100644
--- a/mozilla/xpcom/glue/standalone/nsXPCOMGlue.cpp
+++ b/mozilla/xpcom/glue/standalone/nsXPCOMGlue.cpp
@@ -299,6 +299,9 @@ AppendDependentLib(LibHandleType libHandle)
 static bool
 ReadDependentCB(pathstr_t aDependentLib, bool do_preload)
 {
+    char lib[MAXPATHLEN];
+    if (realpath(aDependentLib, lib))
+        aDependentLib = lib;
     if (do_preload) {
         ReadAheadLib(aDependentLib);
     }
