From: Mike Hommey <mh@glandium.org>
Date: Wed, 24 Feb 2010 18:38:43 +0100
Subject: mozilla-config.h was renamed js-confdefs.h in js/src

---
 mozilla/js/src/Makefile.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mozilla/js/src/Makefile.in b/mozilla/js/src/Makefile.in
index df8251b..f1318e4 100644
--- a/mozilla/js/src/Makefile.in
+++ b/mozilla/js/src/Makefile.in
@@ -751,7 +751,7 @@ distclean::
 
 DIST_GARBAGE = config.cache config.log config.status \
    config/autoconf.mk \
-   unallmakefiles js-config js-config.h mozilla-config.h
+   unallmakefiles js-config js-config.h js-confdefs.h
 
 distclean::
 	cat unallmakefiles | $(XARGS) rm -f
